addon/mixins/form-data-adapter.js

```diff
@@ -13,13 +13,16 @@ export default Ember.Mixin.create({
 
     if (typeof FormData !== 'undefined' && data && this.formDataTypes.contains(type)) {
       var formData = new FormData();
-      var root = Ember.keys(data)[0];
-
-      Ember.keys(data[root]).forEach(function(key) {
-        if (typeof data[root][key] !== 'undefined') {
-          formData.append(root + "[" + key + "]", data[root][key]);
+      var fields = this.getDataToTransform(data);
+
+      Ember.keys(fields).forEach(function(key) {
+        if (typeof fields[key] !== 'undefined') {
+          formData.append(
+            this.getFormKey(key, fields),
+            this.getFormValue(key, fields)
```

Comment:
Could you please clarify, why do we need both the iteration over ``fields`` and passing the whole ``fields`` array to those functions from within the iterations? I.e. couldn't we do something like ``this.getFormKey(key, fields[key])``